Implemented completion aware interface for stecman symfony console completion

JoshuaBehrens opened the pull request:
### 1. Why is this change necessary?
For easier usability of console commands. This is a rebased follow up to #33 

Did you ever recognized a whole uuid? Do you know by heart what the plugin was spelled?

Examples in usage were already shown in the [pull request for shopware 5](https://github.com/shopware/shopware/pull/1488).

### 2. What does this change do, exactly?
Implements for all core/shipped commands the CompletionAwareInterface
Adds hints in the documentation

### 3. Describe each step to reproduce the issue or behaviour.
Press tab to complete your shopware console commands. It just does not work.

### 4. Which documentation changes (if any) need to be made because of this PR?
https://docs.shopware.com should get a hint on the setup and pros.
There is a complementary pull request to https://github.com/shopware/development/pull/20 to ensure this features is ready in the default docker container.

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

htkassner commented:
Thank you for your contribution. Your repository is not available anymore, so I couldn´t import your code. I will close this PR now. Is this still an issue for you in 5.2? Maybe you would like to contribute to 5.2 if this is still relevant.

JoshuaBehrens committed:
Implemented completion aware interface for stecman symfony console completion

shyim commented:
I guess you missed the composer.json require for the stecman package

JoshuaBehrens commented:
Well I made it in the wrong composer.json. It is stuck in the development template 😆

JoshuaBehrens committed:
Add missing dependency

JoshuaBehrens commented:
I added the missing dependency

mitelg commented on composer.json:
```diff
@@ -112,7 +112,8 @@
         "twig/twig": "2.12.3",
-        "zircote/swagger-php": "3.0.3"
+        "zircote/swagger-php": "3.0.3",
+        "stecman/symfony-console-completion": "^0.11.0"
```
please add the requirement to all the submodules `composer.json` files, where this is used in the commands

JoshuaBehrens committed:
Implement completion for theme:change

JoshuaBehrens committed:
Put requirement to every sub-repository

JoshuaBehrens commented:
I added the dependency to the other repositories as well, implemented the feature for theme:change
![expand-theme-change](https://user-images.githubusercontent.com/1133593/77864306-16808080-7228-11ea-995c-75b67a4e2b87.gif)
and did a rebase :)

JoshuaBehrens committed:
Amend docs for new layout

JoshuaBehrens commented:
Rebase is out and I also changed the new docs

PaddyS commented on src/Core/Framework/Migration/Command/CreateMigrationCommand.php:
```diff
@@ -21,21 +28,82 @@ class CreateMigrationCommand extends Command
+
+            return $result;
+        } elseif ($argumentName === 'namespace') {
+            // TODO find good way to determine the namespace
```
Hey @JoshuaBehrens,
will you still take care of that tone? :-)

JoshuaBehrens replied:
Good catch. I think by now it is easier for me to find namespaces :) I'll amend that

pweyck commented:
This PR was closed due to inactivity. If this change is still important to you, feel free to create a new pull request.
For more information about our contribution guidelines, see https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline
